SimCalorimetry/EcalSimAlgos/interface/EcalHitResponse_Ph2.h

```diff
@@ -0,0 +1,137 @@
+#ifndef SimCalorimetry_EcalSimAlgos_EcalHitResponse_Ph2_h
+#define SimCalorimetry_EcalSimAlgos_EcalHitResponse_Ph2_h
+
+#include "CalibFormats/CaloObjects/interface/CaloTSamplesBase.h"
+#include "CalibFormats/CaloObjects/interface/CaloSamples.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHit.h"
+#include "CalibCalorimetry/EcalLaserCorrection/interface/EcalLaserDbService.h"
+#include "DataFormats/Provenance/interface/Timestamp.h"
+
+#include <unordered_map>
+#include <vector>
+
+typedef unsigned long long TimeValue_t;
+
+class CaloVShape;
+class CaloVSimParameterMap;
+class CaloVHitCorrection;
+class CaloVHitFilter;
+class CaloSimParameters;
+class CaloSubdetectorGeometry;
+class CaloVPECorrection;
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+class EcalHitResponse_Ph2 {
```

Comment:
this class only changes the sampling frequency. not sufficient justification to duplicate all the code. can be accomplished by templating, or even just initializing the class with a member that indicates the sample size.

Reply:
We templated EBHitResponse, which inherit from EcalHitResponse, and there is not more necessity of EcalHitResponse_Ph2